### ISSUE TITLE
segmentation violation during startup

### ISSUE THREAD

**Issue by frittenlab** [unpoller/unpoller#442]
Hi,

I have been running unpoller version 2.1.3 for nearly a year now. I noticed that you have been actively working on the repo. And I saw that you published a new container version. So I tried to upgrade my deployment in Kubernetes today. During startup the pod crashed instantly with this error message:

```
2022/11/23 14:28:26 [INFO] Loading Configuration File: /config/unifi-poller.conf
2022/11/23 14:28:26 [INFO] UniFi Poller v Starting Up! PID: 1
2022/11/23 14:28:26 [INFO] Found 1 site(s) on controller https://unifi.XXX: default (Home)
2022/11/23 14:28:26 [INFO] Configured UniFi Controller 1 of 1:
2022/11/23 14:28:26 [INFO]    => URL: https://unifi.XXXX (verify SSL: false)
2022/11/23 14:28:26 [INFO]    => Version: 7.2.95 (717f9a42-1228-4e86-8df7-d111b297e836)
2022/11/23 14:28:26 [INFO]    => Username: unifipoller (has password: true)
2022/11/23 14:28:26 [INFO]    => Hash PII / Poll Sites: false / all
2022/11/23 14:28:26 [INFO]    => Save Sites / Save DPI: true / false (metrics)
2022/11/23 14:28:26 [INFO]    => Save Events / Save IDS: false / false (logs)
2022/11/23 14:28:26 [INFO]    => Save Alarms / Anomalies: false / false (logs)
2022/11/23 14:28:26 [INFO]    => Save Rogue APs: false
2022/11/23 14:28:26 [INFO] Internal web server disabled!
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x9 pc=0x3bad18]

goroutine 66 [running]:
github.com/unpoller/datadogunifi.(*DatadogUnifi).Run(0x0?, {0x702a00?, 0x400021e020?})
	/home/runner/go/pkg/mod/github.com/unpoller/datadogunifi@v0.0.0-20221109161507-2c4fbce23b95/datadog.go:194 +0x58
github.com/unpoller/poller.(*UnifiPoller).runOutputMethods.func1(0x0?)
	/home/runner/go/pkg/mod/github.com/unpoller/poller@v0.0.0-20210623104748-50161c195d5e/outputs.go:85 +0x38
created by github.com/unpoller/poller.(*UnifiPoller).runOutputMethods
	/home/runner/go/pkg/mod/github.com/unpoller/poller@v0.0.0-20210623104748-50161c195d5e/outputs.go:84 +0x100
```

I built a helm chart a while ago to deploy unpoller in my Kubernetes cluster incase you are interested:

https://github.com/frittenlab/helm-charts/tree/main/charts/unifi-poller

**Comment by platinummonkey** [comment]
Hm mind sharing if you have the DataDog plugin enabled or disabled? I couldn’t tell from that chart link.

**Comment by frittenlab** [comment]
No I do not use any additional plugins.

**Comment by mfenniak** [comment]
I've got the same SIGSEGV...


```
Nov 23 10:27:43 anxi unpoller[65381]: 2022/11/23 10:27:43 [INFO] Loading Configuration File: /nix/store/qzd7w7qpzwqljvly9aswiq56y68cz7pk-prometheus-unifi-poller-exporter.js>
Nov 23 10:27:43 anxi unpoller[65381]: 2022/11/23 10:27:43 [INFO] UniFi Poller v Starting Up! PID: 65381
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO] Found 1 site(s) on controller https://unifi.kainnef.com: default (Home)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO] Configured UniFi Controller 1 of 1:
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => URL: https://unifi.kainnef.com (verify SSL: true)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Version: 7.2.95 (c21fadc9-e2d9-41d9-abb8-036efe9c532e)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Username: readonly (has password: true)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Hash PII / Poll Sites: false / all
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Save Sites / Save DPI: true / false (metrics)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Save Events / Save IDS: false / false (logs)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Save Alarms / Anomalies: false / false (logs)
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO]    => Save Rogue APs: false
Nov 23 10:27:44 anxi unpoller[65381]: 2022/11/23 10:27:44 [INFO] Prometheus exported at http://0.0.0.0:9130/ - namespace: unpoller
Nov 23 10:27:44 anxi unpoller[65381]: panic: runtime error: invalid memory address or nil pointer dereference
Nov 23 10:27:44 anxi unpoller[65381]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x9 pc=0xb43e44]
Nov 23 10:27:44 anxi unpoller[65381]: goroutine 16 [running]:
Nov 23 10:27:44 anxi unpoller[65381]: github.com/unpoller/datadogunifi.(*DatadogUnifi).Run(0x1?, {0xeddae8?, 0xc0000aa020?})
Nov 23 10:27:44 anxi unpoller[65381]:         github.com/unpoller/datadogunifi@v0.0.0-20221109161507-2c4fbce23b95/datadog.go:194 +0x44
Nov 23 10:27:44 anxi unpoller[65381]: github.com/unpoller/poller.(*UnifiPoller).runOutputMethods.func1(0xc0000ce000?)
Nov 23 10:27:44 anxi unpoller[65381]:         github.com/unpoller/poller@v0.0.0-20210623104748-50161c195d5e/outputs.go:85 +0x31
Nov 23 10:27:44 anxi unpoller[65381]: created by github.com/unpoller/poller.(*UnifiPoller).runOutputMethods
Nov 23 10:27:44 anxi unpoller[65381]:         github.com/unpoller/poller@v0.0.0-20210623104748-50161c195d5e/outputs.go:84 +0xd0
N
```

With this config file (some paths, urls removed):
```
{
  "influxdb": {
    "disable": true
  },
  "loki": {
    "interval": "2m",
    "pass": "file:///another-secret",
    "tenant_id": "",
    "timeout": "10s",
    "url": "",
    "user": "",
    "verify_ssl": false
  },
  "poller": {
    "debug": false,
    "quiet": false
  },
  "prometheus": {
    "http_listen": "0.0.0.0:9130",
    "report_errors": false
  },
  "unifi": {
    "controllers": [
      {
        "hash_pii": false,
        "pass": "file:///secret.pass",
        "save_alarms": false,
        "save_anomalies": false,
        "save_dpi": false,
        "save_events": false,
        "save_ids": false,
        "save_sites": true,
        "sites": [
          "all"
        ],
        "url": "https://unifi.example.com",
        "user": "readonly",
        "verify_ssl": true
      }
    ]
  }
}
```

**Comment by mfenniak** [comment]
The problem seems to be that the `datadog` config is required and having it missing will cause this segfault.  I've worked around this by having `... datadog: { "disable": true } ...` in my config.

**Comment by platinummonkey** [comment]
I think I've also found a related bug will be able to cut a release as well

**Comment by frittenlab** [comment]
I tried the new version today. It crashed with the same message I reported yesterday. I tried disabling datadog in my config as @mfenniak mentioned. That seems to do the job for the moment. Still a strange issue as datadog disabled true should be the default.

**Comment by platinummonkey** [comment]
okay, I'm going to flip this logic ie explicit enable: true

**Comment by platinummonkey** [comment]
the change to explicit enable was merged in v2.3.0 thanks for reporting!
